misk/src/main/kotlin/misk/inject/Guice.kt

```diff
@@ -12,12 +12,21 @@ import javax.inject.Singleton
 
 internal inline fun <reified T : Any> LinkedBindingBuilder<in T>.to() = to(T::class.java)
 
-@Suppress("UNCHECKED_CAST")
 internal inline fun <reified T : Any> Binder.addMultibinderBinding(): LinkedBindingBuilder<T> {
+    return addMultibinderBinding(null)
+}
+
+internal inline fun <reified T : Any, reified A : Annotation> Binder.addMultibinderBindingWithAnnotation() = addMultibinderBinding<T>(A::class.java)
+
+@Suppress("UNCHECKED_CAST")
+private inline fun <reified T : Any> Binder.addMultibinderBinding(annotation: Class<out Annotation>?): LinkedBindingBuilder<T> {
```

Comment:
can you make `annotation` default to null and save some overloads?

Reply:
Right, that's the Java coming through when I forget about default params.